### PR TITLE
Make CoreContext::Create<T> return the most-specific type possible.

### PR DESCRIPTION
**Pull request by vdods**
This makes CoreContext::Create<T> (and related functions) return std::shared_ptr\<CoreContextT\<T>> instead of just std::shared_ptr<CoreContext>.

- [ ] Wait of victor to give the go

### CODE REVIEW

**Comment by vdods** [comment]
@codemercenary The placement of the `private` method in CoreContext is probably incorrect, but I wasn't sure what the correct place was.  Also, can you review the unit test?  Also, I don't currently know enough about AutoInjectable to write a meaningful unit test for the CoreContext::Create<T> method which accepts an AutoInjectable&&.

**Comment by vdods** [comment]
Hold off on this for the moment.

**Comment by codemercenary** [comment]
Good to go now?

**Comment by vdods** [comment]
If you're ok with the placement of the private method UntypedCreate (adjacent to the public method Create), then it's ready.

File: autowiring/CoreContext.h [hunk 1]
```diff
@@ -573,23 +573,35 @@ class CoreContext:
...
+    return std::make_shared<CoreContextT<T>>(pParent, backReference);
+  }
+
+private:
+  /// \internal
+  /// <summary>
+  /// Creation helper routine
+  /// </summary>
+  template<class T>
+  static std::shared_ptr<CoreContext> UntypedCreate(
```
Comment (codemercenary) [inline autowiring/CoreContext.h]: Prefer the name `CreateUntyped` instead.  This causes all `Create` methods to be near each other in sorted statement completion lists.

Comment (vdods) [inline autowiring/CoreContext.h]: Done.  Good point.

**Pull request merged**